lightning_pose/models/heatmap_tracker.py

```diff
@@ -202,10 +201,26 @@ def create_double_upsampling_layer(
 
     def heatmaps_from_representations(
         self,
-        representations: TensorType["batch", "features", "rep_height", "rep_width"],
+        representations: Union[
+            TensorType["batch", "features", "rep_height", "rep_width"],
+            TensorType["batch", "features", "rep_height", "rep_width", "frames"],
+        ],
     ) -> TensorType["batch", "num_keypoints", "heatmap_height", "heatmap_width"]:
-        """Wrapper around self.upsampling_layers for type and shape assertion."""
+        """Handle context frames then upsample to get final heatmaps."""
+
+        # handle context frames first
+        if (self.mode == "2d" and self.do_context) or self.mode == "3d":
+            # push through a linear layer to get the final representation
+            # input shape (batch, features, rep_height, rep_width, frames)
+            # output shape (batch, features, rep_height, rep_width, 1)
+            representations = self.representation_fc(representations)
+            # final squeeze
+            # output shape (batch, features, rep_height, rep_width)
```

Comment:
maybe instead of comment, can have 
```
representations: TensorType["batch", "features", "rep_height", "rep_width"] = torch.squeeze(representations, 4)
```